### ISSUE TITLE
Naming of mutations and rule 17

### ISSUE THREAD

**Issue by emilebosch** [Shopify/graphql-design-tutorial#5]
Thank you very much for the guide. I do love it a lot, although I have some questions with Rule number 17. Is rule number 17 purely for the autocomplete and ease for developers? If we communicate in the business language for the rest of the API shouldn't this be also the case (And we should solve the editor/grouping problem instead)

It does seem that the api leaks developer preference in the business domain, It does result in a structured api though where things are grouped functionality but it is conflicting with some other API's and result in a little weird naming at times.  Any thoughts?

**Comment by eapache** [comment]
> ease for developers

This is a pretty important thing when building an API.

> If we communicate in the business language for the rest of the API shouldn't this be also the case

We're not changing any vocabulary, just choosing a specific ordering. I still consider this to be using the business language.

> we should solve the editor/grouping problem instead

Sure? But we're not the authors of the GraphQL spec, and the working group has shown no interest in this problem.

**Comment by eapache** [comment]
tl;dr we agree this isn't ideal, but it seems like the best set of trade-offs to us.
